tensorflow/core/kernels/cwise_op_gpu_select.cu.cc

```diff
@@ -33,6 +33,20 @@ struct SelectFunctor<GPUDevice, T> {
 };
 
 template <typename T>
+struct SelectScalarFunctor<GPUDevice, T> {
+  void operator()(const GPUDevice& d, typename TTypes<T>::Flat out,
+                  typename TTypes<bool>::ConstScalar cond,
+                  typename TTypes<T>::ConstFlat then_flat,
+                  typename TTypes<T>::ConstFlat else_flat) {
+    if (cond()){
```

Comment:
Or you could write a custom cuda kernel instead of using Eigen.  That was my suggestion, since there's no complex selecting going on here, it's just 'A or B'.